libs/facilities/backend/timeseries/src/lib/services/timeseries.service.ts

```diff
@@ -0,0 +1,68 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { PrismaService } from 'common-backend-prisma';
+import { catchError, from, map, Observable } from 'rxjs';
+
+import { IGetTimeSeriesParams, IGetTimeseriesQuery } from '../interfaces/request.interface';
+import {
+	ITimeSeriesDataItemResponse,
+	ITimeSeriesItemResponse,
+} from '../interfaces/response.interface';
+
+@Injectable()
+export class XdTimeseriesService {
+	constructor(
+		@Inject(forwardRef(() => PrismaService))
+		private readonly prismaService: PrismaService,
+		// @Inject(ConfigService)
+		// private readonly configService: ConfigService,
+	) {}
+
+	/**
+	 * Get timeseries data based on the entityId and propertySetName
+	 */
+	public getTimeSeriesFromDB(
+		args: IGetTimeSeriesParams & IGetTimeseriesQuery,
+	): Observable<ITimeSeriesDataItemResponse[]> {
+		const { entityId, propertySetName } = args;
+
+		return from(
+			this.prismaService.timeSeriesDataItem.findMany({
+				where: {
+					timeSeriesItementityId: entityId,
+					timeSeriesItempropertySetName: propertySetName,
+					time: {
+						gte: args.from,
+						lte: args.to,
+					},
+				},
+				take: args.limit,
+				orderBy: {
+					time: args.sort,
+				},
+			}),
+		).pipe(
+			map((items) => {
+				return items.map((item) => ({
+					time: item.time,
+					...this.prismaService.selectKeysFromJSON(item.data, args.select),
+				}));
+			}),
+			catchError((err: Error) => {
```

Comment:
Unnecessary (according to ur comment in my PR)

Reply:
Yup :)